src/ArmTemplates/Commands/Executors/ExtractorExecutor.cs

```diff
@@ -164,252 +150,267 @@ public async Task GenerateSplitAPITemplates(FileNames fileNames)
+            if (!apiDictionary.ContainsKey(this.extractorParameters.ApiVersionSetName))
+            {
+                throw new NoApiVersionSetWithSuchNameFoundException("API Version Set with this name doesn't exist");
+            }
+            else
+            {
+                Logger.LogInformation("Start extracting the API version set {0}", this.extractorParameters.ApiVersionSetName);
+
+                foreach (string apiName in apiDictionary[this.extractorParameters.ApiVersionSetName])
+                {
+                    // generate seperate folder for each API
+                    string apiFileFolder = string.Concat(this.extractorParameters.FilesGenerationRootDirectory, $@"/{apiName}");
+                    System.IO.Directory.CreateDirectory(apiFileFolder);
+                    await this.GenerateTemplates(apiFileFolder, apiName, null, null);
+                }
+
+                // create master templates for this apiVersionSet 
+                string versionSetFolder = string.Concat(this.extractorParameters.FilesGenerationRootDirectory, this.extractorParameters.FileNames.VersionSetMasterFolder);
+                System.IO.Directory.CreateDirectory(versionSetFolder);
+                await this.GenerateTemplates(versionSetFolder, null, apiDictionary[this.extractorParameters.ApiVersionSetName], null);
+
+                Logger.LogInformation($@"Finish extracting APIVersionSet {this.extractorParameters.ApiVersionSetName}");
+            }
+        }
+
+        /// <summary>
+        /// Generates templates for multiple specified APIs
+        /// </summary>
+        public async Task GenerateMultipleAPIsTemplates()
+        {
+            if (this.extractorParameters.MultipleApiNames.IsNullOrEmpty())
+            {
+                throw new Exception("multipleAPIs parameter doesn't have any data");
+            }
+
+            Logger.LogInformation("Start extracting multiple APIs (amount is {0} APIs)", this.extractorParameters.MultipleApiNames.Count);
+
+            foreach (string apiName in this.extractorParameters.MultipleApiNames)
+            {
+                // generate seperate folder for each API
+                string apiFileFolder = string.Concat(this.extractorParameters.FilesGenerationRootDirectory, $@"/{apiName}");
+                System.IO.Directory.CreateDirectory(apiFileFolder);
+                await this.GenerateTemplates(apiFileFolder, apiName, null, null);
+            }
+
+            // create master templates for these apis 
+            string groupApiFolder = string.Concat(this.extractorParameters.FilesGenerationRootDirectory, this.extractorParameters.FileNames.GroupAPIsMasterFolder);
+            System.IO.Directory.CreateDirectory(groupApiFolder);
+            await this.GenerateTemplates(groupApiFolder, null, this.extractorParameters.MultipleApiNames, null);
```

Comment:
There are a lot of calls all over the place where we keep passing nulls. Could those parameters be optional and null by default?

Reply:
Great point, fixed it